src/factorization.jl

```diff
@@ -75,20 +75,16 @@ function implicit_restart!(arnoldi::Arnoldi{T}, min = 5, max = 30) where {T}
     H = copy(arnoldi.H)
 
     rotations = ListOfRotations(T, max)
-    # rotations = ListOfRotations(eltype(H),max)
+
     for m = max : -1 : min + 1
 
         shifted_qr_step!(view(H, 1 : m + 1, 1 : m), λs[m], rotations)
         mul!(view(Q, 1 : max, 1 : m), rotations)
 
-        # Qs, Rs = qr(view(H, 1 : m, 1 : m) - λs[m] * I)
-        # h *= Qs[m, m - 1]
-        # Q = view(Q, 1 : max, 1 : m) * Qs
-        # H = Rs * Qs + λs[m] * I
     end
 
-    # Remove the last columns of H
-    H = H[:, 1 : min]
+    # Remove the last columns and rows of H
+    H = H[1 : min+1, 1 : min]
 
     # Update the H[end, end] value
     H[min + 1, min] = h
```

Comment:
And this should be gone as well, cause it's already handled in the implicit restart.